### PR TITLE
improved readability

### PR DESCRIPTION
**Pull request by enmoed**
### Changed
-changed names of classes in _telemetry
-removed 3 functions in _telemetry

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/_token_providers.py [hunk 1]
```diff
@@ -180,9 +177,7 @@ async def _get_token_async():
 
             if token is None:
                 async with self._async_lock:
-                    token = await KustoTracing.call_func_tracing_async(
-                        self._get_token_impl_async, name_of_span=f"{self.name()}.get_token_impl_async", tracing_attributes=context
-                    )
+                    token = await Span.run_async(self._get_token_impl_async, f"{self.name()}.get_token_impl_async", context)
```
Comment (AsafMah) [inline azure-kusto-data/azure/kusto/data/_token_providers.py:180]: I think here and in other palces where it's long it's good to break it up to new lines

File: azure-kusto-data/azure/kusto/data/client.py [hunk 2]
```diff
@@ -277,19 +277,17 @@ def _execute(
...
             json=json_payload,
             data=payload,
             timeout=timeout.seconds,
             stream=stream_response,
             allow_redirects=False,
-            name_of_span="KustoClient.http_post",
-            tracing_attributes=http_trace_attributes,
         )
+        http_trace_attributes = SpanAttributes.create_http_attributes(url=endpoint, method="POST", headers=request_headers)
+        response = Span.run(invoker, name_of_span="KustoClient.http_post", tracing_attributes=http_trace_attributes)
```
Comment (AsafMah) [inline azure-kusto-data/azure/kusto/data/client.py:290]: Here I think you should break it to lines and inline http_trace_attributes (you don't need the param names also)

File: azure-kusto-data/azure/kusto/data/_telemetry.py [hunk 3]
```diff
@@ -97,58 +97,29 @@
...
-        kusto_func: Callable = kusto_trace(func)
-        return kusto_func(*args, **kwargs)
+class Span:
+    """
+    Span class for telemetry
+    """
 
     @staticmethod
-    async def call_func_tracing_async(func: Callable, *args, **kwargs):
+    def run(invoker: Callable, name_of_span: str = None, tracing_attributes=None, kind: str = SpanKind.INTERNAL):
```
Comment (AsafMah) [inline azure-kusto-data/azure/kusto/data/_telemetry.py:106]: Use generics to signify that the Method returns what the callable does, like in the example I showed you in teams.

File: azure-kusto-data/tests/test_telemetry.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import pytest
 
-from azure.kusto.data._telemetry import KustoTracing, KustoTracingAttributes
+from azure.kusto.data._telemetry import Span, SpanAttributes
```
Comment (AsafMah) [inline azure-kusto-data/tests/test_telemetry.py:3]: you don't need to put the tests in the class

File: azure-kusto-data/tests/test_telemetry.py [hunk 5]
```diff
@@ -11,29 +11,70 @@ class TestTelemetry:
...
+    def test_run_valid_invoker():
+        # Happy path test for run method with valid invoker function and name of span
+        def invoker():
+            return "Hello World"
+
+        span = Span.run(invoker, "test_span")
+        assert span is not None
+
+    @staticmethod
+    async def test_run_async_valid_invoker():
```
Comment (AsafMah) [inline azure-kusto-data/tests/test_telemetry.py:32]: async tests need "@pytest.mark.asyncio" or they're not doing anything

**Review by AsafMah — COMMENTED** [review]
Minor comments

File: azure-kusto-data/azure/kusto/data/_telemetry.py (hunk 3 above)
Comment (enmoed) [reply azure-kusto-data/azure/kusto/data/_telemetry.py:106]: addressed

File: azure-kusto-data/tests/test_telemetry.py (hunk 4 above)
Comment (enmoed) [reply azure-kusto-data/tests/test_telemetry.py:3]: removed class

File: azure-kusto-data/tests/test_telemetry.py (hunk 5 above)
Comment (enmoed) [reply azure-kusto-data/tests/test_telemetry.py:32]: thanks

File: azure-kusto-data/tests/test_telemetry.py [hunk 6]
```diff
@@ -11,29 +11,70 @@ class TestTelemetry:
 
     @staticmethod
     def plus_one(num):
-        KustoTracingAttributes.add_attributes(tracing_attributes={"foo": "bar"})
+        SpanAttributes.add_attributes(tracing_attributes={"foo": "bar"})
```
[the remark this answers is not in the document]
Comment (enmoed) [reply azure-kusto-data/tests/test_telemetry.py:14]: removed

File: azure-kusto-data/azure/kusto/data/client.py (hunk 2 above)
Comment (enmoed) [reply azure-kusto-data/azure/kusto/data/client.py:290]: done - removed params from http_trace but left params for MonitoredActivity.invoke

File: azure-kusto-data/azure/kusto/data/_token_providers.py (hunk 1 above)
Comment (enmoed) [reply azure-kusto-data/azure/kusto/data/_token_providers.py:180]: addressed

File: azure-kusto-data/azure/kusto/data/client.py (hunk 2 above)
Comment (enmoed) [reply azure-kusto-data/azure/kusto/data/client.py]: Note that in some cases black. reformats so that it is on the same line

**Review by yogilad — DISMISSED** [review]

**Review by AsafMah — DISMISSED** [review]
Just fix the comment

File: azure-kusto-data/tests/aio/test_async_token_providers.py [hunk 7]
```diff
@@ -247,7 +247,7 @@ async def test_app_key_provider(self):
     @pytest.mark.asyncio
     async def test_app_cert_provider(self):
         # default details are for kusto-client-e2e-test-app
-        # to run the test download the certs from Azure Portal
```
[the remark this answers is not in the document]
Comment (enmoed) [reply azure-kusto-data/tests/aio/test_async_token_providers.py:250]: thanks

**Pull request merged**